File: January/Day16/57_Insert_Interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
    vector<vector<int>>ans;

    for(auto interval : intervals)
    {
       if(interval[1]< newInterval[0])
       {
           ans.push_back(interval);
       } 
       else if(interval[0]>newInterval[1])
       {
           ans.push_back(newInterval);
           newInterval = interval;
       }
       else
       {
           newInterval[0] = min(newInterval[0],interval[0]);
           newInterval[1] = max(newInterval[1],interval[1]);
       }
    }

    ans.push_back(newInterval);
    return ans;
    }
};

File: January/Day16/57_Insert_Interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        ans = []
        for interval in intervals:
            if interval[1] < newInterval[0]:
                ans.append(interval)
            elif newInterval[1] < interval[0]:
                ans.append(newInterval)
                newInterval = interval
            else :
                newInterval[0] = min(interval[0], newInterval[0])
                newInterval[1] = max(interval[1], newInterval[1])
        ans.append(newInterval)
        return ans
